specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1134 @@
+        IVector<Windows.Foundation.Uri> OptionalPackageUris { get; };
+        IVector<Windows.Foundation.Uri> RelatedPackageUris { get; };
+        Windows.Foundation.Uri ExternalLocationUri;
+        Windows.Management.Deployment.StubPackageOption StubPackageOption;
+        Boolean DeveloperMode;
+        Boolean ForceUpdateFromAnyVersion;
+        Boolean InstallAllResources;
+        Boolean RequiredContentGroupOnly;
+        Boolean StageInPlace;
+        Boolean AllowUnsigned;
+
+        Boolean IsExpectedDigestsSupported { get; };            // Requires Windows >= 10.0.22621.0 (aka Win11 22H2)
+        IMap<Windows.Foundation.Uri, String> ExpectedDigests{ get; };
+    }
+
+    // Requires Windows >= 10.0.19041.0 (aka 2004 aka 20H1)
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass RegisterPackageOptions
+    {
+        RegisterPackageOptions();
+
+        PackageVolume AppDataVolume;
+        IVector<Windows.Foundation.Uri> DependencyPackageUris { get; };
+        IVector<String> OptionalPackageFamilyNames { get; };
+        Windows.Foundation.Uri ExternalLocationUri;
+        Boolean DeveloperMode;
+        Boolean ForceAppShutdown;
+        Boolean ForceTargetAppShutdown;
+        Boolean ForceUpdateFromAnyVersion;
+        Boolean InstallAllResources;
+        Boolean StageInPlace;
+        Boolean AllowUnsigned;
+        Boolean DeferRegistrationWhenPackagesAreInUse;
+
+        Boolean IsExpectedDigestsSupported { get; };            // Requires Windows >= 10.0.22621.0 (aka Win11 22H2)
+        IMap<Windows.Foundation.Uri, String> ExpectedDigests{ get; };
+    }
+
+    // Requires Windows >= 10.0.19041.0 (aka 2004 aka 20H1)
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass RemovePackageOptions
+    {
+        RemovePackageOptions();
+
+        Boolean OkIfNotFound;
+        Boolean PreserveApplicationData;
+        Boolean PreserveRoamableApplicationData;
+        Boolean RemoveForAllUsers;
+    }
+
+    // Requires Windows >= 10.0.22000.0 (aka Win11 aka 21H2 aka SV1)
+    [feature(Feature_PackageManager)]
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass ProvisionPackageOptions
+    {
+        ProvisionPackageOptions();
+
+        IVector<String> OptionalPackageFamilyNames { get; };
+        IVector<String> ProjectionOrderPackageFamilyNames { get; };
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass EnsurePackageIsReadyOptions
+    {
+        EnsurePackageIsReadyOptions();
+
+        AddPackageOptions AddPackageOptions;
+    }
+
+    [contract(PackageDeploymentContract, 1)]
+    runtimeclass PackageDeploymentManager
+    {
+        // Get an instance of the manager
+        static PackageDeploymentManager GetDefault();
+
+        //-------------------------------------------------------------
+        // IsReady
+
+        // Return true if the package(s) are present and available for use
+
+        Boolean IsPackageReady(String package);
```

Comment:
This doesn't have a `By`, so I expect it to take a Package` as the parameter, not a string. What is the string? Package Family Name? Package Full Name? Aumid?